R/default-internal-plot-configuration.R

```diff
@@ -0,0 +1,322 @@
+# class definition ---------------------------------
+
+#' @title Convenience class for plot configuration for OSP plots
+#'
+#' @description
+#'
+#' An intermediary R6 class that provides a single portal to prepare the
+#' `PlotConfiguration` object needed by `{tlf}` plotting functions. The specific
+#' type of `PlotConfiguration` objects can access all necessary objects formed
+#' from user inputs using this internal object.
+#'
+#' @field labels `tlf::LabelConfiguration` object defining properties of labels.
+#' @field legend `tlf::LegendConfiguration` object defining legend properties.
+#' @field xAxis `tlf::XAxisConfiguration` object defining `x`-axis properties.
+#' @field yAxis `tlf::YAxisConfiguration` object defining `y`-axis properties.
+#' @field background `tlf::BackgroundConfiguration` object defining the
+#'   configuration of background.
+#' @field lines `tlf::ThemeAestheticSelections` object or list defining how
+#'   lines are plotted.
+#' @field points `tlf::ThemeAestheticSelections` object or list defining how
+#'   points are plotted.
+#' @field ribbons `tlf::ThemeAestheticSelections` object or list defining
+#'   how ribbons are plotted.
+#' @field errorbars `tlf::ThemeAestheticSelections` object or list defining
+#'   how errorbars are plotted.
+#' @field export `tlf::SaveConfiguration` object defining saving properties.
+#'
+#' @keywords internal
+DefaultInternalPlotConfiguration <- R6::R6Class(
```

Comment:
I am not quite sure I understand the need for this object. It seems to be just a data older from one to the other. Probably a mapper method c`reateTimeProfileConfig` etc.. would do the trick as well. But if this is easier to understand, we can keep this

Reply:
Wasn't it the role of the DefaultPlotConfiguration
For some reasons, it seems that we are mapping not once, but twice
DefaultPlotConfig => DEfaultInternalPlotConfig => TlfPlotConfig

Reply:
Alright. Got rid of `DefaultInternalPlotConfiguration`!

So now it's only: `DefaultPlotConfig => TlfPlotConfig`